### ISSUE TITLE
GPRS SIM wil not setup

### ISSUE THREAD

**Issue by Bjorntenbroeke** [Limon93/Linkit-ONE-Traccar-Client#1]
Hi, 

the sode wil stop at the SIM setup. 
It there any reason why it is setup with LSMS?

```
  // GSM setup
	while(!LSMS.ready()){
  		delay(1000);
  		Serial.println("Insert SIM");
  	}
    	Serial.println("SIM ready.");	
  	//pinMode(LEDGPS, OUTPUT);
  	
```

This is the replay I am getting from the serial monitor.
The SIM card is in there and working. 
Also for other code it works fine. 

16:58:32.166 -> Limon93Studio Traccar client.
16:58:32.166 -> GPS powered ON.
16:58:33.182 -> Insert SIM
16:58:34.162 -> Insert SIM
16:58:35.177 -> Insert SIM
16:58:36.193 -> Insert SIM
16:58:37.173 -> Insert SIM
16:58:38.189 -> Insert SIM

**Comment by Limon93** [comment]
Hi,
This code (at least the sim initialization) should work as it is part of offical examples, maybe a recent update of firmware, sdk, board made this initialization fail.
Are you using an updated version of board+arduino sdk?
Can you link a code working for your board+sim configuration? I need more info to help you.

Regards,
limon93.

**Comment by Bjorntenbroeke** [comment]
Hi, 

Thank you for your replay. Really hoping to get it running :-)

I have tested it once by using a older version on my Macbook. This worked once. 
There the SIM worked, and it got until the getting the GPS and trying to send it to my server. 

When I wanted to change the IP and APN, it got dropped again. 

Because it was giving me INSERT SIM I was on the route of testing my SIM card. 

With the code below I can send a SMS to the number, and receive it. 
So for that I now that the SIM it self is working. 

It would be really great to get this running, because I am looking at it for 2 days now and not getting forward....... 
Harder for me is also checking the Traccar side, because I have my Traccar running on the Synology it is really impossible to check if there is comming "something" in.

```
#include <LGSM.h>

void setup() {
  // put your setup code here, to run once:
  Serial.begin(9600);
  
  while(!LSMS.ready())
    delay(1000);
  Serial.println("SIM ready for work!");
}

void loop() {
  // put your main code here, to run repeatedly:
  char buf[20];
  int v;
  if(LSMS.available())
  {
    Serial.println("There is a new message.");
    
    LSMS.remoteNumber(buf, 20);
    Serial.print("Number:");
    Serial.println(buf);
    
    Serial.print("Content:");
    while(true)
    {
      v = LSMS.read();
      if(v < 0)
        break;
      Serial.print((char)v);
    }
    Serial.println();
    
    LSMS.flush();
  }
  delay(1000);
}
```

**Comment by Limon93** [comment]
Can you tell me your APN? it is not sensible data i guess.
Are you writing it correctly in the #define section?
Are you setting up APN user and password? I asked your apn to verify if it requires user and pass.
Ah, you got a 2G or 3G sim? In the 3G case this board is no luck for you.
Check in your phone if you have data connection in "Only gprs mode" or "Only edge mode".
SMS functionality and data functionality may rely on different radio bands, that is maybe why you can send sms but no data.
Let me know.

**Comment by Bjorntenbroeke** [comment]
Hi, 

Sorry for my late reaction, I was on the road yesterday. 

I have two SIM Cards whom I test with. 

Both with data roaming. Both work in my Phone and have SMS/Internet connection. 
I am able to open websites, send SMS and make a phone call. 

KPN (Netherlands). Here I set the APN to "internet" (no username and password)
Vodafone (Germany) Here I set it to "web.vodafone.de" also no username and password.

As I understood from different forums is that the settings I see on my phone are the Network setting of the phone itself. Turning off/on the Only GPRS mode should be a device setting. As I understood. 

I just called the Dutch KPN. They told me that gprs is the number is standard on. 

On both: no code for sim (both disabled).
